### PR TITLE
DRAFT PROPOSAL: Bulk ops

### PR DESCRIPTION
**Pull request by sleekslush**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -182,33 +182,35 @@ export abstract class BaseStorage {
...
-    if (this.hasExtensionApi) {
-      const dataMap = await this.#primaryClient.get(key)
+  protected rawGet = async (key: string): Promise<string | undefined> => {
+    return await this.rawGetMany([key])[key]
+  }
 
-      return dataMap[key]
+  protected rawGetMany = async (keys: string[]): Promise<Record<string, string | undefined>> => {
+    if (this.hasExtensionApi) {
+      return this.#primaryClient.get(keys)
```
Comment (louisgv) [inline src/index.ts:191]: @sleekslush Does this work in Firefox?

**Comment by louisgv** [comment]
Can we also add getItems / setItems / removeItems alias? Those might be useful :-?

File: src/index.test.ts
Comment (louisgv) [inline src/index.test.ts:1]: Let's add a test for the bulk operation (similar to the one in #48)

File: src/index.ts (hunk 1 above)
Comment (sleekslush) [reply src/index.ts:191]: Yea, both chrome and firefox support arrays of keys

**Comment by sleekslush** [comment]
> Can we also add getItems / setItems / removeItems alias? Those might be useful :-?

We can just rename the ones I specified to this instead. I have no feelings towards it

**Comment by louisgv** [comment]
> > Can we also add getItems / setItems / removeItems alias? Those might be useful :-?
> 
> We can just rename the ones I specified to this instead. I have no feelings towards it

SGTM 👍

**Comment by faahim** [comment]
This would be super useful! Not sure if this is blocked on something?

**Comment by louisgv** [comment]
@faahim blocked on when @sleekslush can finish the PR lol... but I think anyone can take on this idea to finalize it too :d

**Pull request closed without merging**